ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py

```diff
@@ -1382,6 +1382,36 @@ def test_tensorflow_unsorted_segment_sqrt_n(
     )
 
 
+# unsorted_segment_prod
+@handle_frontend_test(
+    fn_tree="tensorflow.math.unsorted_segment_prod",
+    data=helpers.array_values(dtype=ivy.int32, shape=(5, 6), min_value=1, max_value=9),
+    segment_ids=helpers.array_values(
+        dtype=ivy.int32, shape=(5,), min_value=0, max_value=4
```

Comment:
The ids could be either `int32` or `int64`, and we should test for all supported cases without hardcoding the `shape`. 
To make sure the size and values of the `segment_ids` match the `data` array, you will probably need to generate these in a `st.composite` helper function.